datasource.go

```diff
@@ -9,6 +9,7 @@ import (
 // DataSource represents a Grafana data source.
 type DataSource struct {
 	ID     int64  `json:"id,omitempty"`
+	UID    string `json:"uid"`
```

Comment:
It doesn't seem to be an attribute used in the POST call. I think it should be omitted if empty
```suggestion
	UID    string `json:"uid,omitempty"`
```